src/arrayMethodJoin.js

```diff
@@ -5,7 +5,22 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const sep = separator === undefined ? ',' : separator;
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i > 0) {
```

Comment:
```suggestion
      if (i !== this.length - 1) {
```

if it's not the last index